FilePersistence/src/version_control/ListMergeComponent.cpp

```diff
@@ -172,17 +172,30 @@ LinkedChangesTransition ListMergeComponent::run(std::shared_ptr<GenericTree>& tr
 				for (auto elemId : chunk->spanBase_)
 				{
 					auto oldNode = treeBase->find(elemId);
-					auto newNode = treeA->find(elemId);
+					auto newNodeA = treeA->find(elemId);
+					auto newNodeB = treeB->find(elemId);
+					auto adjustChange = [&](GenericNode* oldNode, GenericNode* newNode,
+							ChangeDependencyGraph& cdgA,
+							ChangeDependencyGraph& cdgB)
+					{
+						auto newChange = std::make_shared<ChangeDescription>(oldNode, newNode);
+						updateChange(newChange, cdgA, cdgB);
+						auto delChange = cdgA.changes().value(elemId);
+						if (delChange)	cdgA.remove(delChange);		// Remove previous change
+						cdgA.insert(newChange);
+						cdgA.recordDependencies(newChange, true);
+						cdgA.recordDependencies(newChange, false);
+					};
+
 					Q_ASSERT(oldNode);
-					Q_ASSERT(newNode);
-					auto newChange = std::make_shared<ChangeDescription>(oldNode, newNode);
-					updateChange(newChange, cdgA, cdgB);
-					newChange->print();
-					auto delChange = cdgA.changes().value(elemId);
-					cdgA.remove(delChange);		// Remove previous change
-					cdgA.insert(newChange);
-					cdgA.recordDependencies(newChange, true);
-					cdgA.recordDependencies(newChange, false);
+					Q_ASSERT(newNodeA || newNodeB);
+					if (newNodeA)	//Add change in cdgA
```

Comment:
Improve the style here:
- no need for the comment, it is obvious
- no need for { and }, it's just a single short statement
- same for the else part.
- make this just on two lines ` if...` and `else ...`